neqo-transport/src/connection.rs

```diff
@@ -334,7 +355,34 @@ impl Connection {
         local_addr: SocketAddr,
         remote_addr: SocketAddr,
     ) -> Res<Self> {
-        let dcid = ConnectionId::generate_initial();
+        let dcid = ConnectionId::generate_initial(true);
+        let local_cids = vec![cid_manager.borrow_mut().generate_cid()];
+        let mut c = Self::new(
+            Role::Client,
+            Client::new(server_name)?.into(),
+            cid_manager,
+            None,
+            protocols,
+            Some(Path {
+                local: local_addr,
+                remote: remote_addr,
+                local_cids,
+                remote_cid: dcid.clone(),
+            }),
+        );
+        c.crypto.create_initial_state(Role::Client, &dcid)?;
+        Ok(c)
+    }
+
+    #[cfg(test)]
+    pub fn new_client_fixed_len_dcid(
```

Comment:
It would really be nice to not have to deal with packets varying in lengths in tests. I'd like to either fix at 8 for tests, or if we remove the non-test variability, then this is not needed.

Reply:
removed